Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -342,44 +342,66 @@ namespace AZ::DocumentPropertyEditor
                     false);
                 return;
             }
-
-            auto containerAttribute = attributes.Find(AZ::Reflection::DescriptorAttributes::Container);
-            if (!containerAttribute.IsNull())
+            else
             {
-                auto container = AZ::Dom::Utils::ValueToTypeUnsafe<AZ::SerializeContext::IDataContainer*>(containerAttribute);
-                m_containers.SetValue(m_builder.GetCurrentPath(), BoundContainer{ container, access.Get() });
-                size_t containerSize = container->Size(access.Get());
-                if (containerSize == 1)
-                {
-                    m_builder.Label("1 element");
-                }
-                else
+                AZStd::string_view labelAttribute = "MISSING_LABEL";
+                Dom::Value label = attributes.Find(Reflection::DescriptorAttributes::Label);
+                if (!label.IsNull())
                 {
-                    m_builder.Label(AZStd::string::format("%zu elements", container->Size(access.Get())));
+                    if (!label.IsString())
+                    {
+                        AZ_Warning("DPE", false, "Unable to read Label from property, Label was not a string");
+                    }
+                    else
+                    {
+                        labelAttribute = label.GetString();
+                    }
                 }
 
-                if (!container->IsFixedSize())
+                auto containerAttribute = attributes.Find(AZ::Reflection::DescriptorAttributes::Container);
+                if (!containerAttribute.IsNull())
                 {
-                    m_builder.BeginPropertyEditor<Nodes::ContainerActionButton>();
-                    m_builder.Attribute(Nodes::ContainerActionButton::Action, Nodes::ContainerAction::AddElement);
-                    m_builder.AddMessageHandler(m_adapter, Nodes::ContainerActionButton::OnActivate.GetName());
-                    m_builder.EndPropertyEditor();
+                    auto container = AZ::Dom::Utils::ValueToTypeUnsafe<AZ::SerializeContext::IDataContainer*>(containerAttribute);
+                    m_containers.SetValue(m_builder.GetCurrentPath(), BoundContainer{ container, access.Get() });
+                    size_t containerSize = container->Size(access.Get());
+                    if (containerSize == 1)
+                    {
+                        m_builder.Label(AZStd::string::format("%s (1 element)", labelAttribute.data()));
+                    }
+                    else
+                    {
+                        m_builder.Label(AZStd::string::format("%s (%zu elements)", labelAttribute.data(), container->Size(access.Get())));
+                    }
```

Comment:
Use `QObject::tr` on the pre-formatted string to enable localization. Something like
```
m_builder.Label(AZStd::string::format(QObject::tr("%s (1 element)").toStdString().c_str(), labelAttribute.data()));
```

Reply:
Nice one. I'll get that change in the next update to this PR.

Reply:
Actually, this is in ReflectionAdapter and it doesn't do much with Qt. Let me rethink that one.

Reply:
yeah, we can tr it in the view. Need to be a little careful if we just happen to hit an existing literal name... let's discuss this Monday